tests/test_class_db.h

```diff
@@ -31,12 +31,800 @@
+		exposed_class.base = ClassDB::get_parent_class(class_name);
+
+		// Add properties
+
+		List<PropertyInfo> property_list;
+		ClassDB::get_property_list(class_name, &property_list, true);
+
+		Map<StringName, StringName> accessor_methods;
+
+		for (const List<PropertyInfo>::Element *E = property_list.front(); E; E = E->next()) {
+			const PropertyInfo &property = E->get();
+
+			if (property.usage & PROPERTY_USAGE_GROUP || property.usage & PROPERTY_USAGE_SUBGROUP || property.usage & PROPERTY_USAGE_CATEGORY) {
+				continue;
+			}
+
+			PropertyData prop;
+			prop.name = property.name;
+			prop.setter = ClassDB::get_property_setter(class_name, prop.name);
+			prop.getter = ClassDB::get_property_getter(class_name, prop.name);
+
+			if (prop.setter != StringName()) {
+				accessor_methods[prop.setter] = prop.name;
+			}
+			if (prop.getter != StringName()) {
+				accessor_methods[prop.getter] = prop.name;
+			}
+
+			bool valid = false;
+			prop.index = ClassDB::get_property_index(class_name, prop.name, &valid);
+			TEST_FAIL_COND(!valid, "Invalid property: '" + exposed_class.name + "." + String(prop.name) + "'.");
+
+			exposed_class.properties.push_back(prop);
+		}
+
+		// Add methods
+
+		List<MethodInfo> virtual_method_list;
+		ClassDB::get_virtual_methods(class_name, &virtual_method_list, true);
+
+		List<MethodInfo> method_list;
+		ClassDB::get_method_list(class_name, &method_list, true);
+		method_list.sort();
+
+		for (List<MethodInfo>::Element *E = method_list.front(); E; E = E->next()) {
+			const MethodInfo &method_info = E->get();
+
+			int argc = method_info.arguments.size();
+
+			if (method_info.name.empty()) {
+				continue;
+			}
+
+			MethodData method;
+			method.name = method_info.name;
+
+			if (method_info.flags & METHOD_FLAG_VIRTUAL) {
+				method.is_virtual = true;
+			}
+
+			PropertyInfo return_info = method_info.return_val;
+
+			MethodBind *m = method.is_virtual ? nullptr : ClassDB::get_method(class_name, method_info.name);
+
+			method.is_vararg = m && m->is_vararg();
+
+			if (!m && !method.is_virtual) {
+				TEST_FAIL_COND(!virtual_method_list.find(method_info),
+						"Missing MethodBind for non-virtual method: '" + exposed_class.name + "." + method.name + "'.");
+
+				// A virtual method without the virtual flag. This is a special case.
+
+				// The method Object.free is registered as a virtual method, but without the virtual flag.
+				// This is because this method is not supposed to be overridden, but called.
+				// We assume the return type is void.
+				method.return_type.name = r_context.names_cache.void_type;
+
+				// Actually, more methods like this may be added in the future, which could return
+				// something different. Let's put this check to notify us if that ever happens.
+				if (exposed_class.name != r_context.names_cache.object_class || String(method.name) != "free") {
+					WARN_PRINT("Notification: New unexpected virtual non-overridable method found."
```

Comment:
Use INFO?

Reply:
I think this could be converted to doctest's `WARN`.

Reply:
`WARN` alone doesn't register those messages if you don't run with `--success` switch, but `WARN_FALSE_MESSAGE` works, which also allows to check the condition above.